File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/domain/entities/events/IEntityChangeEventHelper.cs
namespace Volo.Abp.Domain.Entities.Events;

/**
 * Used to trigger entity change events.
*/
public interface IEntityChangeEventHelper
{
    void PublishEntityCreatingEvent(Object entity);
    void PublishEntityCreatedEvent(Object entity);

    void PublishEntityUpdatingEvent(Object entity);
    void PublishEntityUpdatedEvent(Object entity);

    void PublishEntityDeletingEvent(Object entity);
    void PublishEntityDeletedEvent(Object entity);
}

File: ueueo-framework/ueueo-domain-driven-design/src/main/java/com/ueueo/ddd/domain/entities/events/IEntityChangeEventHelper.java
package com.ueueo.ddd.domain.entities.events;

/**
 * Used to trigger entity change events.
 */
public interface IEntityChangeEventHelper {
    void publishEntityCreatedEvent(Object entity);

    void publishEntityUpdatedEvent(Object entity);

    void publishEntityDeletedEvent(Object entity);
}
